modules/extension/wms/src/main/java/org/geotools/map/WMSCoverageReader.java

```diff
@@ -356,8 +365,11 @@ ReferencedEnvelope initMapRequest(ReferencedEnvelope bbox, int width, int height
         // for some silly reason GetMapRequest will list the layers in the opposite order...
         List<Layer> reversed = new ArrayList<Layer>(layers);
         Collections.reverse(reversed);
+        List<String> rStyles = new ArrayList<>(styles);
+        Collections.reverse(rStyles);
+        int i=0;
```

Comment:
Hum... maybe just a "for i" loop then?  Or else, the layers structure could be replaced with a List<Map.Entry<Layer, String>> or a custom bean holding both layer and style.